### DISCUSSION TITLE
Any plans to support multiplier of futures?

### DISCUSSION THREAD

**Discussion by gansaihua** [kernc/backtesting.py#448] [Ideas]
Hi - I'd like to incorporate multiplier of futures, e.g. the ES futures contract has a multiplier of $50.

**Comment by kernc** [comment]
Can you model this sufficiently with [`Backtest(..., margin=)`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest)?

**Reply by gansaihua** [reply]
Thank you for your amazing work. I am just getting started with it.  

There seems to be an workaround, use margin = 50 / 1 (which cannot pass the margin check). As I see it, margin is used for leverage, but multiplier is used in scaling the shares.  

I think this might work: 
1) add a parameter "mult = 1" to both Backtest and Data constructors, 
2) add " * self.__broker._data._mult" to pl property of Trade class, 
3) add mult to liquidity check in broker._process_orders,
4) add mult to precompute size in broker._process_orders

How do you like it?

**Reply by kernc** [reply]
I don't. 

> e.g. the ES futures contract has a multiplier of $50.

Isn't this basically the same as buying some `size` times 50? As the underlying asset moves $1, your position moves $50.

**Reply by fortesantonio** [reply]
In the case of futures "mult" is usefull to have as a parameter for the leverage brokerages offer to clients on certain assets, but most importantly as an alternative method to account for "pip cost" which is the main variable for calculating pnl in the futures market.

**Reply by fortesantonio** [reply]
Pip Value = (Stop Loss Length in Pips) * (Pip Cost)
Position Size in Lots = (% of Account Risk) / (Pip Value)
Limit Profit in Pips= (Stop Loss Length in Pips) * (Reward to Risk Multiple) 

Stop Loss in $ Value = (Pip Cost) * (Lots) * (Stop Loss Length in Pips)
Limit Profit in $ Value = (Pip Cost) * (Lots) * (Limit Profit Length in Pips)

Applied Leverage = (Live Asset Price) * (Position Size in Lots) / (Account Balance)


https://www.fxcm.com/markets/insights/how-is-the-value-of-a-pip-determined/
https://www.dailyfx.com/education/beginner/what-is-a-pip.html

**Reply by fortesantonio** [reply]
"Margin" in futures can be calculated as a percentage of the financial asset (usually around 2%) or as a fixed currency rate per contract and it's value is substracted from the account's "Equity" as "Used Margin" to determine the "Used Margin %", "Usable Margin" and "Usable Margin %".

**Comment by kernc** [comment]
One can model this sufficiently with [`Backtest(..., margin=)`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest).

From [Investopedia](https://www.investopedia.com/ask/answers/what-do-sp-500-dow-and-nasdaq-futures-contracts-represent/):

> The E-mini Nasdaq multiplier is 20, worth $20 per point, while the E-mini S&P 500 carries a 50 multiplier that's worth $50 per point. If the Dow falls 100 points, then the buyer will lose about $500 while a short seller will gain around $500. 
> [...]
> For example, the Nasdaq 100 ended at 9,555.52 [...] and all three indexes went up precisely 1%. Then, an E-mini Nasdaq 100 contract would rise by around $1,911, [...]

9555.52 * 20 / 100 = 1911.10 :ok_hand: 

Which should produce the same result as running the backtest with `margin=1/20`. If you want to incorporate additional _leverage_, simply account for it in margin:
```py
futures_multiple = 20
leverage = 100

bt = Backtest(..., margin=1/futures_multiple * 1/leverage)
